test/session_tests/DocumentQueryTest.ts

```diff
@@ -24,222 +28,181 @@ describe('Document query test', () => {
     ({defaultDatabase, defaultUrl} = (this.currentTest as IRavenObject));
   });
 
-  beforeEach((done: MochaDone) => {
-    store = DocumentStore.create(defaultUrl, defaultDatabase);
-    store.initialize();
-
-    const indexMap: string = [
-      'from doc in docs ',
-      'select new {',
-      'name = doc.name,',
-      'key = doc.key,',
-      'doc_id = doc.key+"_"+doc.name}'
-    ].join('');
-
-
-    const indexSort: IndexDefinition = new IndexDefinition('Testing_Sort', indexMap, null, {
-      fields: {
-        "key": new IndexFieldOptions(SortOptions.Numeric),
-        "doc_id": new IndexFieldOptions(null, null, true)
-      }
-    });
-
-    store.getRequestsExecutor()
-      .execute(new PutIndexesCommand(indexSort))
-      .then((): BluebirdPromise.Thenable<object> => {
-        session = store.openSession();
-
-        return BluebirdPromise.all([
-          session.store(session.create({name: 'test101', key: 2, order: 'a'}, 'product'), 'products/101'),
-          session.store(session.create({name: 'test10', key: 3, order: 'test'}, 'product'), 'products/10'),
-          session.store(session.create({name: 'test106', key: 4, order: 'c'}, 'product'), 'products/106'),
-          session.store(session.create({name: 'test107', key: 5, order: null}, 'product'), 'products/107'),
-          session.store(session.create({name: 'test107', key: 6, order: null}, 'product'), 'products/103'),
-          session.store(session.create({name: 'new_testing', key: 90, order: 'd'}, 'product'), 'products/108'),
-          session.store(session.create({
-            name: 'testing_order',
-            key: 92,
-            product_id: 'products/108'
-          }, 'order'), 'orders/105'),
-          session.store(session.create({
-            name: 'withNesting',
-            product: {name: 'testing_order', key: 4, order: null}
-          }, 'company'), 'company/1')
-        ])
-        .then(() => done());
-      });
+  beforeEach(async () => {
+    store = DocumentStore.create(defaultUrl, defaultDatabase).initialize();
+    session = store.openSession();
+
+    const requestsExecutor: RequestsExecutor = store.getRequestsExecutor();
+    const productsTestingSort: ProductsTestingSort = new ProductsTestingSort(requestsExecutor);
+
+    await productsTestingSort.execute();
+    await session.store<Product>(session.create<Product>(new Product('products/101', 'test101', 2, 'a')));
```

Comment:
@mobilunity-user I'm not sure I follow. We just wanted to have generic load and query methods. I can't recall anything about create(). Could you elaborate please?